File: cookbook/python/automations/endpoints/create.py
from dotenv import load_dotenv
from retab import Retab

assert load_dotenv("../../../.env.local")

reclient = Retab()

print("making a query to create an endpoint")
automation = reclient.processors.automations.endpoints.create(
    name="Endpoint Automation 2",
    processor_id="proc_o4dtLxizT0kDAjeKuyVLA",
    webhook_url="https://your-server.com/webhook",
)

print(automation.model_dump_json(indent=2))

File: cookbook/typescript/automations/endpoints/create.ts
import { config } from 'dotenv';
import { AsyncRetab } from '@retab/node';



async function main() {
  // Load environment variables  
  config({ path: '../../../.env.local' });
  
  const reclient = new AsyncRetab();
  
  console.log("making a query to create an endpoint");
  const automation = await reclient.processors.automations.endpoints.create({
    name: "Endpoint Automation 2",
    processor_id: "proc_o4dtLxizT0kDAjeKuyVLA",
    webhook_url: "https://your-server.com/webhook",
  });
  
  console.log(JSON.stringify(automation, null, 2));
  
}

main().catch(console.error);